src/services/exchange/exchange.service.ts

```diff
@@ -41,14 +44,29 @@ export class ExchangeService {
     return toExchangeRate / fromExchangeRate;
   }
 
-  private async getExchangeResult(): Promise<ExchangeResult> {
+  async getFiatCodes(): Promise<string[]> {
+    const fiatCodesResult: FiatCodesExchangeResult =
+      (await this.getExchangeResult('symbols')) as FiatCodesExchangeResult;
+
+    return Object.keys(fiatCodesResult.symbols);
+  }
+
+  private async getExchangeResult(
+    path?: string,
+  ): Promise<ExchangeResult | FiatCodesExchangeResult> {
     const baseUrl = this.configurationService.get<string>('exchange.baseUri');
     const apiKey = this.configurationService.get<string>('exchange.apiKey');
+    const url = path ? `${baseUrl}/${path}` : baseUrl;
 
     try {
-      const { data } = await this.networkService.get(baseUrl, {
+      const { data } = await this.networkService.get(url, {
         params: { access_key: apiKey },
       });
+
+      if (!data?.success) {
+        throw new InternalServerErrorException('Bad response from Exchange');
+      }
+
       return data;
     } catch (error) {
       this.httpErrorHandler.handle(error);
```

Comment:
Also, by extracting the function to an `ExchangeClient` we could also create a specific `ErrorHandler` for the Exchange errors, since their payload is sightly different from our Safe services error payloads. 🤔

Reply:
Regarding this, I think we could create a general application-wide error handler which parser all the errors and defines a common payload interface. So every data provider kind should have an error handler which maps specific errors to the general error format.

Reply:
The discussion around error codes might also be helpful for this one. Where we have a generic error case but a specific error code that helps us identify the source of the issue.

Also Nest already sets some generic error handler. For this PR I wouldn't say it's expected to "set" the appropriate messages and error codes. I think if you remove our error handler usage here and just throw the error it would be fine at this stage 👍